website/blog/2024-06-18-monitor-ferretdb-performance-using-coroot.md

```diff
@@ -0,0 +1,136 @@
+
+With Coroot, you can visualize every aspect of your entire setup.
+That includes all its components, including your applications and their subsystems, databases, and, most importantly, how they all communicate.
+
+Coroot heavily relies on eBPF, a technology that enhances observability and monitoring by eliminating the need to instrument application code manually.
+Custom programs can run directly within the Linux kernel without changing kernel source code or loading additional kernel modules.
+You get deeper and more extensive information on system calls, network functions, and security issues – just better observability.
+
+## Setting up Coroot for FerretDB monitoring
+
+[Refer to the Coroot documentation](https://coroot.com/docs) to set up Coroot and all its components on Docker or via Helm.
+
+In this guide, we will set up Coroot to monitor our FerretDB instance.
+If you are yet to setup FerretDB, see the [FerretDB Docker installation guide](https://docs.ferretdb.io/quickstart-guide/docker/).
+
+Deploy Coroot with the following command:
+
+```sh
+curl -fsS https://raw.githubusercontent.com/coroot/coroot/main/deploy/docker-compose.yaml | docker compose -f - up -d
+```
+
+Since Coroot is deployed locally, you can access it at http://localhost:8080/.
+
+Depending on your setup, you may need to modify the Docker compose `yaml` file and configure Prometheus to pick up FerretDB metrics.
+
+To update the Prometheus container's configuration file, `exec` into the container:
+
+```sh
+docker exec -it <prometheus_container_name> /bin/sh
+```
+
+Then navigate to the file to edit:
+
+```sh
+cd /etc/prometheus
+vi prometheus.yml
+```
+
+Add the following to the Prometheus configuration file:
+
+```text
+scrape_configs:
+  - job_name: 'ferretdb'
+    metrics_path: '/debug/metrics'
+    static_configs:
+      - targets: ['<ferretdb-container>:8088']
+```
+
+Ensure to replace `<ferretdb-container>` with the FerretDB container name or IP address and then restart the Prometheus container so it can pick up the new configuration.
+
+```sh
+docker restart <prometheus_container_name>
+```
+
+Comfirm that the FerretDB metrics are being collected by Prometheus by navigating to the Prometheus targets at http://localhost:9090/targets.
+
+![Prometheus targets](/img/blog/ferretdb-coroot/prometheus-targets.png)
+
+Once the setup is complete, Coroot will start collecting metrics from FerretDB.
+
+## Monitor FerretDB performance with Coroot
+
+The Coroot dashboard provides the full details on all components.
+
+Ensure Prometheus integration is correctly set up to collect metrics from FerretDB.
+
+![Coroot prometheus integration](/img/blog/ferretdb-coroot/prometheus-integration.png)
+
+From the Coroot dashboard, you can view the FerretDB metrics through Prometheus.
+
+![CPU dashboard 1](/img/blog/ferretdb-coroot/cpu-metrics-1.png)
+
+![CPU dashboard 1](/img/blog/ferretdb-coroot/cpu-metrics-2.png)
```

Comment:
Those graphs show Prometheus requests to FerretDB to gather metrics, not FerretDB metrics